### PR TITLE
#2907 jump back in search groups

### PR DESCRIPTION
**Pull request by electricRGB**
- Adds search groups to Jump Back In section. :)

### CODE REVIEW

File: Client/Frontend/Home/FxHomeJumpBackInView.swift [hunk 1]
```diff
@@ -83,13 +83,21 @@ class FxHomeJumpBackInCollectionCell: UICollectionViewCell {
...
-        if let item = viewModel.jumpableTabs[safe: indexPath.row] {
+        if indexPath.row == (viewModel.jumpList.itemsToDisplay - 1),
+           let group = viewModel.jumpList.group {
+
+            cell.itemTitle.text = group.searchTerm.localizedCapitalized
+            cell.itemDetails.text = String(format: .FirefoxHomeJumpBackInSectionGroupSiteCount, group.groupedItems.count)
+            cell.heroImage.image = UIImage(imageLiteralResourceName: "recently_closed").withRenderingMode(.alwaysTemplate)
+
+        } else {
+            let item = viewModel.jumpList.tabs[indexPath.row]
```
Comment (nishant2718) [inline Client/Frontend/Home/FxHomeJumpBackInView.swift:100]: nit: maybe you can add `[safe: indexPath.row]` here?

File: Client/Frontend/Home/FxHomeJumpBackInView.swift [hunk 2]
```diff
@@ -109,8 +116,13 @@ extension FxHomeJumpBackInCollectionCell: UICollectionViewDataSource {
...
 extension FxHomeJumpBackInCollectionCell: UICollectionViewDelegate {
     func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
-        if let tab = viewModel.jumpableTabs[safe: indexPath.row] {
-            viewModel.switchTo(tab)
+        if indexPath.row == viewModel.jumpList.itemsToDisplay - 1,
+           let group = viewModel.jumpList.group {
+            viewModel.switchTo(group: group)
+
+        } else {
+            let tab = viewModel.jumpList.tabs[indexPath.row]
```
Comment (nishant2718) [inline Client/Frontend/Home/FxHomeJumpBackInView.swift:124]: nit: `[safe: ]` here as well

**Review by nishant2718 — APPROVED** [review]
Very minor nits, but this is 🔥 !

File: Client/Frontend/Home/FxHomeJumpBackInView.swift (hunk 1 above)
Comment (electricRGB) [reply Client/Frontend/Home/FxHomeJumpBackInView.swift:100]: I'm happy to make this change, and all for it, but I don't understand what it brings to the table? If the cell count is based on the number of items in the jumpList, then `index.row` can never be greater than the number of items. So, then, is the `safe` subscript not redundant and without meaningful purpose?

Comment (nishant2718) [reply Client/Frontend/Home/FxHomeJumpBackInView.swift:100]: Yeah that's true - in that case we can skip these.

**Comment by electricRGB** [comment]
Merged without BR as it's failing on a test it often fails on.

**Comment by SimonBasca** [comment]
@electricRGB is this PR pointing to the correct issue? I think it's pointing to a different PR in github? 
Should the fix be for this issue? https://mozilla-hub.atlassian.net/browse/FXIOS-2907

**Pull request merged**